mantle/platform/qemu.go

```diff
@@ -306,6 +306,61 @@ func (inst *QemuInstance) SwitchBootOrder() (err error) {
 	return nil
 }
 
+// RemovePrimaryBlockDevice deletes the primary device from a qemu instance
+// and sets the seconday device as primary. It expects that all block devices
+// are mirrors.
+func (inst *QemuInstance) RemovePrimaryBlockDevice() error {
+	var primaryDevice string
+	var secondaryDevicePath string
+	monitor, err := newQMPMonitor(inst.tempdir)
+	if err != nil {
+		return errors.Wrapf(err, "Could not connect to QMP device")
+	}
+	if err := monitor.Connect(); err != nil {
+		return errors.Wrapf(err, "Could not connect to QMP device")
+	}
+
+	defer func() {
+		e := monitor.Disconnect()
+		if err != nil {
+			err = fmt.Errorf("%v; %v", err, e)
+		} else {
+			err = e
+		}
+	}()
+	blkdevs, err := listQMPBlkDevices(monitor, inst.tempdir)
+	if err != nil {
+		return errors.Wrapf(err, "Could not list block devices through qmp")
+	}
+	for _, dev := range blkdevs.Return {
+		if !dev.Removable {
+			if dev.Inserted.BackingfileDepth == 1 {
+				primaryDevice = dev.DevicePath
+			} else {
```

Comment:
WDYT about adding an additional check here that the `DevicePath` at least looks like it is a `virtio-backend` instead of marking the `secondaryDevicePath` as the last block device that isn't removable & doesn't have a `BackingFileDepth` of `1`?

Reply:
Just wondering if this condition (https://github.com/coreos/coreos-assembler/pull/1880/files#diff-16a92b89e4fdab63ff64b808b2495d53843ca8e5b6614bb5493be3077c8726ccR349) handles the scenario mentioned in your comment.

Reply:
Ah yeah, my mistake.